packages/generate/src/edgeql-js/generateInterfaces.ts

```diff
@@ -200,11 +194,17 @@ export const generateInterfaces = (params: GenerateInterfacesParams) => {
     plainTypesExportBuf.decreaseIndent();
     plainTypesExportBuf.writeln([t`};`]);
 
-    if (wrapInNamespace) {
-      plainTypesCode.decreaseIndent();
-      plainTypesCode.writeln([t`}`]);
-      plainTypesCode.writeln([js`}`]);
-      plainTypesCode.addExport(module.internalName, { modes: ["js"] });
+    plainTypesCode.decreaseIndent();
+    plainTypesCode.writeln([t`}`]);
+    plainTypesCode.addExport(module.internalName, { modes: ["js"] });
+
+    if (module.isRoot && module.name === "default") {
+      for (const [typeName, typeRef] of module.types) {
+        plainTypesCode.writeln([`import ${typeName} = ${typeRef};`]);
+      }
+      plainTypesCode.writeln([
+        `export {${[...module.types.keys()].join(", ")}};`,
+      ]);
```

Comment:
Seems like this is already covered by the existing tests in `interfaces.test.ts`, but do you think it'd be easy to write a quick check for the pathological case we're fixing here?